Server/mods/deathmatch/logic/luadefs/CLuaTeamDefs.cpp

```diff
@@ -67,282 +65,86 @@ void CLuaTeamDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Team", "Element");
 }
 
-int CLuaTeamDefs::SetPlayerTeam(lua_State* luaVM)
+std::variant<CTeam*, bool> CLuaTeamDefs::CreateTeam(lua_State* lua, const std::string_view name, const std::uint8_t red, const std::uint8_t green, const std::uint8_t blue) noexcept
 {
-    CPlayer* pPlayer;
-    CTeam*   pTeam;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pPlayer);
-    argStream.ReadUserData(pTeam, NULL);
+    CLuaMain& vm = lua_getownercluamain(lua);
+    CResource* resource = vm.GetResource();
 
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetPlayerTeam(pPlayer, pTeam))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    if (!resource)
+        return false;
 
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
+    std::string string(name);
```

Comment:
Just change the parameter type from `std::string_view` to `std::string`, no need for this.